worlds/sm64ex/Rules.py

```diff
@@ -58,6 +58,10 @@ def set_rules(world,player,area_connections):
         add_rule(world.get_location("BoB: Mario Wings to the Sky", player), lambda state: state.has("Cannon Unlock BoB", player))
         add_rule(world.get_location("WF: Blast Away the Wall", player), lambda state: state.has("Cannon Unlock WF", player))
         add_rule(world.get_location("JRB: Blast to the Stone Pillar", player), lambda state: state.has("Cannon Unlock JRB", player))
+    if (world.AreaRandomizer[player].value):
```

Comment:
I don't think this helps, as it will still be able to have the course in a different floor. I'm already working on a fix, so I'd say you can remove those lines

Reply:
The timer on the clock will continue to tick down while moving from vanilla TTC to actual TTC though, right? So I don't see how this helps

Reply:
No? I'm not quite sure how you mean, or how you've got the clock state implemented if the player hasn't jumped into the clock painting before loading the actual TTC stage (wherever it's located).
Correct me if I'm wrong, but from my understanding, it seems that the clock defaults to the "normal" speed, otherwise use the speed that should have been loaded by jumping into the clock painting. 
Clock speed does not matter for the most part, just mainly for the Red Coins. If not frozen, it's a massive pain to do. Best case imo would be setting the clock speed to "frozen" when selecting the red coin star. That way, time will be stopped, and you can do star 6, and when you select another star, time will be moving again

Reply:
The way i'm handling the course rando and TTC time atm is quite simple, I do nothing and hope it works out lol
I've looked at the relevant code now and it seems that you're right. However, as I'm adding a different, game-side fix soon I don't see the need to add this atm.
(For reference, I'm adding a clock to whichever painting contains TTC)